### PR TITLE
Peggy's first MongoDB

### PR DESCRIPTION
**Pull request by pwangy**
Link to my project: https://project-sweden-covid19-data.herokuapp.com/

Reflections: This week I continued working with my own data set and wanted to see if I could use a unique key from one collection to get data from the other. It took a while to get to this point and a good next step would to look into how to use custom _ids or other variables besides the mongo generated unique ids.

I would like to learn about how to do this if you have any guidance on that. 😌

Other than that, I am so happy to be done with this assignment and am really looking forward to unplugging for a bit to rest. Happy holidays and see you in 2021! 🎉

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -3,36 +3,98 @@ import bodyParser from 'body-parser'
...
+    covid19Data.forEach(async reportItem => {
+      const newReport = new Report({
+        ...reportItem,
+        regionNum: instanceArray.find(key => key.regionId === reportItem.regionNum)
+      })
+      await newReport.save()
+    })
+  }
+  seed()
+}
```
Comment (sandra-shumanteva-kaevska) [inline server.js:53]: Good job Peggy 👍

File: server.js [hunk 2]
```diff
@@ -3,36 +3,98 @@ import bodyParser from 'body-parser'
...
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+// middleware to detect if db is available and throws error to user if it is not.
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+   next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable'})
```
Comment (sandra-shumanteva-kaevska) [inline server.js:66]: It is very good to show an error with correct status 👍

File: server.js [hunk 3]
```diff
@@ -3,36 +3,98 @@ import bodyParser from 'body-parser'
...
+    if (region) {
+      const regionalReport = await Report.find({ regionNum: mongoose.Types.ObjectId(region.id) })
+      res.json(regionalReport)
+    } else {
+      res.status(404).json({ error: 'Region not found' })
+    } 
+  } catch(err) {
+      res.status(400).json({ error: 'Invalid region id'})
+    }
 })
```
Comment (sandra-shumanteva-kaevska) [inline server.js:95]: Showing two more errors :) Good job 👍

**Review by sandra-shumanteva-kaevska — APPROVED** [review]
Great job Peggy!!. You had succeeded  with your project to use a unique key from one collection to get data from the other. 
Just continue with this great work :D.

**Review by KarolCislowski — APPROVED** [review]
Nice Project 👍🏻

**Pull request closed without merging**